src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,149 @@
+package duke;
+
+import duke.data.Storage;
+import duke.data.TaskList;
+import duke.io.Command;
+import duke.io.Parser;
+import duke.io.Ui;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Main program managing all main functions.
+ */
+public class Duke {
+    private final Storage storage;
+    private TaskList taskList;
+    private final Ui ui;
+
+    /**
+     * Constructs the main Duke object to run the program.
+     *
+     * @param filePath The path of the save file to be used/created
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            taskList = storage.loadTaskData();
+        } catch (IOException e) {
+            ui.showLoadingError();
+            taskList = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the main program loop.
+     * Manages the main loop, handling all user input and saving data/etc.
+     */
+    public void run() {
+        String userInput;
+        boolean isTaskListUpdated = false;
+        try (Scanner sc = new Scanner(System.in)) {
+            while (true) {
+                try {
+                    if (isTaskListUpdated) {
+                        // update file
+                        storage.saveTaskData(taskList);
+                        isTaskListUpdated = false;
+                    }
+
+                    userInput = Parser.getInput(sc);
+
+                    Command command = Parser.parse(userInput);
+                    String description;
+                    String keyword;
+                    LocalDate dateTime;
+                    int index;
+
+                    switch (command.getCommand()) {
```

Comment:
I think this is not the correct switch statement style, the cases should be aligned with the switch statement

Reply:
I like how you indented the code to make the logic clear, nice! However, I did observe that your imports are in the wrong order (it should be java first , _then_ your own created ones.